src/analysisd/decoders/decoder.h

```diff
@@ -79,18 +82,36 @@ typedef struct dbsync_context_t {
  * list and to get the first osdecoder
  */
 void OS_CreateOSDecoderList(void);
-int OS_AddOSDecoder(OSDecoderInfo *pi, OSDecoderNode **pn_osdecodernode, OSDecoderNode **npn_osdecodernode);
+int OS_AddOSDecoder(OSDecoderInfo *pi, OSDecoderNode **pn_osdecodernode,
+                    OSDecoderNode **npn_osdecodernode, OSList* log_msg);
 OSDecoderNode *OS_GetFirstOSDecoder(const char *pname);
 int getDecoderfromlist(const char *name);
 char *GetGeoInfobyIP(char *ip_addr);
-int SetDecodeXML(void);
+
+/**
+ * @brief Read decoder files and save them in the decoder list.
+ * @param log_msg list to save log messages.
+ * @return int 0 in case of error, 1 otherwise.
+ */
+int SetDecodeXML(OSList* log_msg);
+
 void HostinfoInit(void);
 int fim_init(void);
 void RootcheckInit(void);
 void SyscollectorInit(void);
 void CiscatInit(void);
 void WinevtInit(void);
 void SecurityConfigurationAssessmentInit(void);
-int ReadDecodeXML(const char *file);
+
+/**
+ * Add decoders to main list
+ * 
+ * @param file path of the decoder configuration xml file.
+ * @param log_msg List to save log messages.
+ * @return 1 Decoder was added to the list.
```

Comment:
When documenting return values, this is the best option:
```c
/*
 * @retval 1 Decoder was added to the list.
 * @retval 2 0 In case of error.
 * @retval 3 File XML_LDECODER (...)
 */